store/src/java/com/zimbra/cs/db/DbMailItem.java

```diff
@@ -5483,4 +5479,104 @@ public static void setPreviousFolder(Mailbox mbox, int id, String prevFolders) t
             DbPool.closeStatement(stmt);
         }
     }
+
+    /**
+     * return Ids of all the emails assosiated for a user in a mailbox.These will be populated in the mail page as the user logs in.
+     *
+     * @param mbox
+     * @param senderEmail
+     * @param isUpdated
+     * @param conn
+     * @return
+     * @throws ServiceException
+     */
+    public static List<Integer> getAllRecordsForSentAndRecieved(Mailbox mbox, String senderEmail, boolean isUpdated)
+            throws ServiceException {
+        PreparedStatement preparedStatement = null;
+        ResultSet rs = null;
+        DbConnection conn = null;
+        try {
+             conn = mbox.getOperationConnection();
+            List<Integer> recordIds = new ArrayList<>();
+            String WHERE_CLAUSE_CHECK_UPDATED = "";
+            if (isUpdated) {
+                WHERE_CLAUSE_CHECK_UPDATED = "AND LOCATOR NOT LIKE '%@@%'";
+            }
+             preparedStatement = conn.prepareStatement(" SELECT id FROM " + getMailItemTableName(mbox,
+                    false) + " where ((sender like concat('%',?,'%')) or (recipients like concat('%',?,'%')))  and folder_id in (2,3,4,5,6,9) " + WHERE_CLAUSE_CHECK_UPDATED);
+            preparedStatement.setString(1, senderEmail);
+            preparedStatement.setString(2, senderEmail);
+             rs = preparedStatement.executeQuery();
```

Comment:
Please re-check the formatting of this file

Reply:
fixed